pkg/security/probe/resolvers/tc_resolver.go

```diff
@@ -0,0 +1,172 @@
+func (tcr *TCResolver) SetupNewTCClassifierWithNetNSHandle(device model.NetDevice, netnsHandle *os.File, m *manager.Manager) error {
+	tcr.Lock()
+	defer tcr.Unlock()
+
+	var combinedErr multierror.Error
+	for _, tcProbe := range probes.GetTCProbes() {
+		// make sure we're not overriding an existing network probe
+		deviceKey := NetDeviceKey{IfIndex: device.IfIndex, NetNS: device.NetNS, NetworkDirection: tcProbe.NetworkDirection}
+		_, ok := tcr.programs[deviceKey]
+		if ok {
+			continue
+		}
+
+		newProbe := tcProbe.Copy()
+		newProbe.CopyProgram = true
+		newProbe.UID = probes.SecurityAgentUID + device.GetKey()
+		newProbe.IfIndex = int(device.IfIndex)
+		newProbe.IfIndexNetns = uint64(netnsHandle.Fd())
+		newProbe.IfIndexNetnsID = device.NetNS
+		newProbe.KeepProgramSpec = false
+		newProbe.TCFilterPrio = tcr.config.NetworkClassifierPriority
+		newProbe.TCFilterHandle = netlink.MakeHandle(0, tcr.config.NetworkClassifierHandle)
+
+		netnsEditor := []manager.ConstantEditor{
+			{
+				Name:  "netns",
+				Value: uint64(device.NetNS),
+			},
+		}
+
+		if err := m.CloneProgram(probes.SecurityAgentUID, newProbe, netnsEditor, nil); err != nil {
+			_ = multierror.Append(&combinedErr, fmt.Errorf("couldn't clone %s: %v", tcProbe.ProbeIdentificationPair, err))
+		} else {
+			tcr.programs[deviceKey] = newProbe
+		}
+	}
+	return combinedErr.ErrorOrNil()
+}
+
+// flushNetworkNamespace thread unsafe version of FlushNetworkNamespace
+func (tcr *TCResolver) FlushNetworkNamespaceID(namespaceID uint32, m *manager.Manager) {
+	tcr.Lock()
+	defer tcr.Unlock()
+
+	for tcKey, tcProbe := range tcr.programs {
+		if tcKey.NetNS == namespaceID {
+			_ = m.DetachHook(tcProbe.ProbeIdentificationPair)
+			delete(tcr.programs, tcKey)
+		}
+	}
+}
+
+// FlushInactiveProbes detaches and deletes inactive probes. This function returns a map containing the count of interfaces
+// per network namespace (ignoring the interfaces that are lazily deleted).
+func (tcr *TCResolver) FlushInactiveProbes(m *manager.Manager, isLazy func(string) bool) map[uint32]int {
+	tcr.Lock()
+	defer tcr.Unlock()
+
+	probesCountNoLazyDeletion := make(map[uint32]int)
+
+	var linkName string
+	for tcKey, tcProbe := range tcr.programs {
+		if !tcProbe.IsTCFilterActive() {
+			_ = m.DetachHook(tcProbe.ProbeIdentificationPair)
+			delete(tcr.programs, tcKey)
+		} else {
+			link, err := tcProbe.ResolveLink()
+			if err == nil {
+				linkName = link.Attrs().Name
+			} else {
+				linkName = ""
+			}
+			// ignore interfaces that are lazily deleted
+			if link.Attrs().HardwareAddr.String() != "" && !isLazy(linkName) {
+				probesCountNoLazyDeletion[tcKey.NetNS]++
+			}
+		}
+	}
+
+	return probesCountNoLazyDeletion
+}
+
+func (tcr *TCResolver) GetTCProbeForKey(key NetDeviceKey) *manager.Probe {
+	tcr.RLock()
+	defer tcr.RUnlock()
+
+	tcProbe, ok := tcr.programs[key]
+	if ok {
+		return tcProbe
+	}
+
+	key.NetworkDirection = manager.Ingress
```

Comment:
I think we should move all the logic in `ResolveNetworkDeviceIfName` (in model.go) to the resolver and rename the `GetTCProbeForKey` accordingly. It feels weird for this function to return either an egress or an ingress probe, and since we're simply resolving the interface name, better have the logic here that in an event resolver function.